pkg/simulation/setup.go

```diff
@@ -343,3 +344,66 @@ func (s *Simulation) handleEnergy() {
 			Write("energy_frame", s.C.F+f)
 	}
 }
+
+func (s *Simulation) handleHurt() {
+	//hurt once interval=300 amount=1,300 element=phys #once at frame 300 (or nearest)
+	if s.cfg.Hurt.Active && s.cfg.Hurt.Once {
+		f := s.cfg.Hurt.Start
+		amt := s.cfg.Hurt.Min + s.C.Rand.Float64()*(s.cfg.Hurt.Max-s.cfg.Hurt.Min)
+		s.cfg.Hurt.Active = false
+
+		ai := combat.AttackInfo{
+			ActorIndex: s.C.Player.Active(),
+			Abil:       "Hurt",
+			AttackTag:  attacks.AttackTagNone,
+			ICDTag:     attacks.ICDTagNone,
+			ICDGroup:   attacks.ICDGroupDefault,
+			StrikeType: attacks.StrikeTypeDefault,
+			Durability: 0,
+			Element:    s.cfg.Hurt.Element,
+			FlatDmg:    amt,
+		}
+		ap := combat.NewSingleTargetHit(s.C.Combat.Player().Key())
+		ap.SkipTargets[targets.TargettablePlayer] = false
+		ap.SkipTargets[targets.TargettableEnemy] = true
+		ap.SkipTargets[targets.TargettableGadget] = true
+
+		s.C.QueueAttack(ai, ap, -1, f) // -1 to avoid snapshot
+
+		s.C.Log.NewEventBuildMsg(glog.LogHurtEvent, -1, "hurt queued (once)").
+			Write("last", s.cfg.Hurt.LastHurt).
+			Write("cfg", s.cfg.Hurt).
+			Write("amt", amt).
+			Write("hurt_frame", s.C.F+f)
```

Comment:
nit: should move to function since code between `every` and `once` is basically the same